packages/auth/src/get-session.ts

```diff
@@ -0,0 +1,28 @@
+import type {
+  GetServerSidePropsContext,
+  NextApiRequest,
+  NextApiResponse,
+} from "next";
+import { unstable_getServerSession } from "next-auth";
+
+import { getAuthOptions } from "./auth-options";
+
+export const getServerSession = async (
+  ctx:
+    | {
+        req: GetServerSidePropsContext["req"];
+        res: GetServerSidePropsContext["res"];
+      }
+    | { req: NextApiRequest; res: NextApiResponse }
+) => {
+  const authOptions = getAuthOptions({
+    // FIXME: How do we get the env's here? Do we even need them, or are empty strings ok?
```

Comment:
see above, not sure if 1. we need this function at all? will we use JWT? or 2. if the secrets actually matter for grabbing the session from the headers, i think just empty string will do, but will take a look at nextauth's src to be sure

Reply:
I'm not quite sure how the JWT strategy works.

I'd prefer persisting in database, but does the use case of validating the JWT in request headers work without the jwt strategy?

Reply:
I've done some more digging on this and it should be fine to use both jwt and session with database persistance. you also don't need every single provider's secrets to get the session, but the secret is needed. 

of course it's not a strict requirement to have the auth package "secret-agnostic" but it's nice to keep it as clean as possible I guess?

Reply:
So that means that the `.env` var in packages/auth needs the `NEXTAUTH_SECRET` but not the provider secrets (that's only for nextjs, which also depends on the `NEXTAUTH_SECRET` though)